app/controllers/v0/sessions_controller.rb

```diff
@@ -136,22 +136,32 @@ def handle_completed_slo
       session         = Session.find(logout_request&.token)
       user            = User.find(session&.uuid)
 
+      # see comments for this method.
+      destroy_user_session!(user, session, logout_request)
       errors = build_logout_errors(logout_response, logout_request, session, user)
 
       if errors.size.positive?
         extra_context = { in_response_to: logout_response&.in_response_to }
         log_message_to_sentry("SAML Logout failed!\n  " + errors.join("\n  "), :error, extra_context)
         redirect_to Settings.saml.logout_relay + '?success=false'
       else
-        logout_request.destroy
-        session.destroy
-        user.destroy
         redirect_to Settings.saml.logout_relay + '?success=true'
```

Comment:
I'd even be happy if we got rid of the conditional altogether, e.g.
```ruby
if errors.size.positive?
  extra_context = { in_response_to: logout_response&.in_response_to }
  log_message_to_sentry("SAML Logout failed!\n  " + errors.join("\n  "), :error, extra_context)
  redirect_to Settings.saml.logout_relay + '?success=false'
end
redirect_to Settings.saml.logout_relay + '?success=true'
```

Reply:
@U-DON does the FE use this in SLO flow at all from what you're seeing? I think since we're changing the logic to destroy a session and user regardless of errors or not, we might not need to show anything on the FE and can just continue to log the behavior on BE for now. 

I strongly suspect a double-click race condition as the reason for some of those failures anyway.

Reply:
The F/E currently uses `success=true` to clear `sessionStorage`, redirect to the home page, and close the pop-up. But it doesn't use `success=false`.

Reply:
And the `success=false` case is actually causing the logging out pop-up to just hang.